Mathlib/Topology/UniformSpace/Separated.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2025 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.Set.Pairwise.Basic
+import Mathlib.Topology.UniformSpace.Defs
+
+/-!
+# Uniform separation
+
+This file defines a notion of separation of a set relative to an entourage.
+
+For an entourage `U`, a `U`-separated set `s` is a set such that every pair of elements of `s` is
+`U`-far.
+
+The concept of uniformly separated sets is used to define two further notions of separation:
+* Metric separation: `Metric.IsSeparated`, defined using the distance entourage.
+* Dynamical nets: `Dynamics.IsDynNetIn`, defined using the dynamical entourage.
+
+## TODO
+
+* Actually use `UniformSpace.IsSeparated` to define the above two notions.
+* Link to the notion of separation given by pairwise disjoint balls.
+-/
+
+open Set
+
+namespace UniformSpace
+variable {X : Type*} {U V : Set (X × X)} {s t : Set X} {x : X}
+
+/-- Given a uniformity `U`, a set `s` is `U`-separated if its elements are pairwise `U`-far from
+each other. -/
+def IsSeparated (U : Set (X × X)) (s : Set X) : Prop := s.Pairwise fun x y ↦ (x, y) ∉ U
+
+protected lemma IsSeparated.empty : IsSeparated U (∅ : Set X) := pairwise_empty _
+protected lemma IsSeparated.singleton : IsSeparated U {x} := pairwise_singleton ..
+
+@[simp] lemma IsSeparated.of_subsingleton (hs : s.Subsingleton) : IsSeparated U s := hs.pairwise _
+
+alias _root_.Set.Subsingleton.uniformSpaceIsSeparated := IsSeparated.of_subsingleton
+
+nonrec lemma IsSeparated.anti_left (hUV : U ⊆ V) (hs : IsSeparated V s) : IsSeparated U s :=
+  hs.mono' fun _x _y hxy h ↦ hxy <| hUV h
```

Comment:
I think the consensus in the naming scheme is that, when there's only one obvious monotonicity, the name should contain `mono` even if its actually antitone (see [Filter.Tendsto.mono_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Tendsto.html#Filter.Tendsto.mono_left) and [Filter.Tendsto.mono_right](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Tendsto.html#Filter.Tendsto.mono_right)).

Reply:
Aha, that's funny how different our experiences are. I always try "mono" first because I know the kind of statement I'm looking for but I don't want to spend time thinking whether it counts as a monotonicity or antitonicity (mostly when there's propositions involved: it takes me a second to remember which way the order goes on `Prop`). So if it's not named "mono" I usually don't find it until later!